### PR TITLE
Flare pouch is a holster now.

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
If there is a flare gun in your flare pouch, quick equip/left click will always draw the flare gun, and not a flare.

https://user-images.githubusercontent.com/66163761/231369564-987495d4-fc58-4bcb-be7e-0c63f9db1121.mp4

image
image

## Why It's Good For The Game
QOL
## Changelog
:cl: @Xander3359 @KillerOrcaCora 
refactor: Flare pouch is a holster now
imageadd: New flare pouch sprites
/:cl:

### CODE REVIEW

**Comment by Xander3359** [comment]
> It probably should be a holster, instead of just imitating holster behavior.

You want me to re-path it? All the loadouts will break again.

**Comment by Lumipharon** [comment]
> > It probably should be a holster, instead of just imitating holster behavior.
> 
> You want me to re-path it? All the loadouts will break again.

It'll delete the invalid item out of the loadout, put people can just edit loadouts now so thats like a 5 second fix.

**Comment by Xander3359** [comment]
> > > It probably should be a holster, instead of just imitating holster behavior.
> > 
> > 
> > You want me to re-path it? All the loadouts will break again.
> 
> It'll delete the invalid item out of the loadout, put people can just edit loadouts now so thats like a 5 second fix.

It breaks item representation which causes it to runtime.

**Comment by Lumipharon** [comment]
me changing item representation itself breaks it, changing items does not. It'll just exclude it when loading the loadout, and resave it without it.

**Review by Kuro020 — APPROVED** [review]

**Comment by Lumipharon** [comment]
Now that I think about it, I'm not actually sure about this change.

If I'm holding a gun, I can't reload a flaregun, so I would prefer to be drawing actual flares.

If I'm using a flaregun, I need 2 free hands so I'm using the draw to draw spare flares more than the flaregun itself.

File: code/game/objects/items/storage/holsters.dm [hunk 1]
```diff
@@ -348,3 +354,39 @@
...
+	refill_types = list(/obj/item/storage/box/m94)
+	refill_sound = "rustle"
+	holsterable_allowed = list(/obj/item/weapon/gun/grenade_launcher/single_shot/flare/marine)
+
+/obj/item/storage/holster/flarepouch/full/Initialize()
+	. = ..()
+	var/obj/item/flare_gun = new /obj/item/weapon/gun/grenade_launcher/single_shot/flare/marine(src)
+	INVOKE_ASYNC(src, PROC_REF(handle_item_insertion), flare_gun)
+	for(var/i in 1 to (storage_slots-flare_gun.w_class))
+		new /obj/item/explosive/grenade/flare(src)
```
Comment (Lumipharon) [inline code/game/objects/items/storage/holsters.dm:383]: put this under all the flarepouch code, since its a child type.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/holsters.dm (hunk 1 above)
Comment (Xander3359) [reply code/game/objects/items/storage/holsters.dm:383]: Simply have the init on /obj/item/storage/holster/flarepouch/init? Sure.

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**